ykshim_client/src/lib.rs

```diff
@@ -151,6 +151,17 @@ impl<I> CompiledTrace<I> {
     /// Execute the trace with the given interpreter context.
     pub unsafe fn execute(&self, ctx: &mut I) -> *mut RawStopgapInterpreter {
         let f = mem::transmute::<_, fn(&mut I) -> *mut RawStopgapInterpreter>(self.ptr());
+        Self::exec_trace(f, ctx)
+    }
+
+    /// Call the compiled trace code.
+    /// This is separate from `execute()` so as to make it easy to get a GDB break point
+    /// immediately before our trace code. It is also named in such a way that `b exec_trace` will
+    /// break at all possilbe entry points to trace code (there is another one in `MTThread`).
+    fn exec_trace(
```

Comment:
```suggestion
    #[inline(never)]
    fn exec_trace(
```

Reply:
I think that may help to prevent the symbol disappearing, but ensures the performance hit?

Reply:
Alternatively it could be

```suggestion
    #[cfg_attr(debug_assertions, inline(never))]
    fn exec_trace(
```

Reply:
Perhaps it would be better as: `#[cfg_attr(not(debug_assertions), inline(always)]`.

The symbol only goes missing if it's annotated `#[inline(always)]`.

Reply:
If I duplicate the function, I can have both.

Reply:
Without `#[inline(always)]` it is still possible to inline it, which loses the symbol.